src/language-js/printer-estree.js

```diff
@@ -5918,14 +5919,30 @@ function printComment(commentPath, options) {
       return "/*" + comment.value + (isInsideFlowComment ? "*-/" : "*/");
     }
     case "CommentLine":
-    case "Line":
+    case "Line": {
+      const isInJSXAttribute = commentPath.getNode(1).type === "JSXAttribute";
+
+      const isLineBeforeEmpty = isPreviousLineEmpty(
+        options.originalText,
+        comment,
+        options
+      );
+
+      // Preserves blank lines above comments to be consistent with our general empty lines rules.
```

Comment:
Not just blank lines above comments, but blank lines in general among JSX props? Also, could you add tests without comments?

Reply:
Good catch. I've added tests and a check for blank lines in general among JSX props.